cmd/soroban-cli/src/signer.rs

```diff
@@ -97,121 +163,118 @@ pub fn sign_soroban_authorizations(
-            ScVal::Bytes(
-                signer
-                    .verifying_key()
-                    .to_bytes()
-                    .to_vec()
-                    .try_into()
-                    .map_err(Error::Xdr)?,
+        let map = ScMap::sorted_from(vec![
+            (
+                ScVal::Symbol(ScSymbol("public_key".try_into()?)),
+                ScVal::Bytes(address.0.to_vec().try_into()?),
             ),
-        ),
-        (
-            ScVal::Symbol(ScSymbol("signature".try_into()?)),
-            ScVal::Bytes(
-                signature
-                    .to_bytes()
-                    .to_vec()
-                    .try_into()
-                    .map_err(Error::Xdr)?,
+            (
+                ScVal::Symbol(ScSymbol("signature".try_into()?)),
+                ScVal::Bytes(signature.try_into()?),
             ),
-        ),
-    ])
-    .map_err(Error::Xdr)?;
-    credentials.signature = ScVal::Vec(Some(
-        vec![ScVal::Map(Some(map))].try_into().map_err(Error::Xdr)?,
-    ));
-    credentials.signature_expiration_ledger = signature_expiration_ledger;
-    auth.credentials = SorobanCredentials::Address(credentials.clone());
-    Ok(auth)
+        ])?;
+        credentials.signature = ScVal::Vec(Some(vec![ScVal::Map(Some(map))].try_into()?));
+        auth.credentials = SorobanCredentials::Address(credentials.clone());
+
+        Ok(auth)
+    }
 }
 
-pub fn sign_tx(
-    key: &ed25519_dalek::SigningKey,
-    tx: &Transaction,
-    network_passphrase: &str,
-) -> Result<TransactionEnvelope, Error> {
-    let tx_hash = hash(tx, network_passphrase)?;
-    let tx_signature = key.sign(&tx_hash);
-
-    let decorated_signature = DecoratedSignature {
-        hint: SignatureHint(key.verifying_key().to_bytes()[28..].try_into()?),
-        signature: Signature(tx_signature.to_bytes().try_into()?),
-    };
+fn hash(network_passphrase: &str) -> xdr::Hash {
+    xdr::Hash(Sha256::digest(network_passphrase.as_bytes()).into())
+}
 
-    Ok(TransactionEnvelope::Tx(TransactionV1Envelope {
-        tx: tx.clone(),
-        signatures: [decorated_signature].try_into()?,
-    }))
+pub struct LocalKey {
+    key: ed25519_dalek::SigningKey,
+    prompt: bool,
 }
 
-pub fn hash(tx: &Transaction, network_passphrase: &str) -> Result<[u8; 32], xdr::Error> {
-    let signature_payload = TransactionSignaturePayload {
-        network_id: Hash(Sha256::digest(network_passphrase).into()),
-        tagged_transaction: TransactionSignaturePayloadTaggedTransaction::Tx(tx.clone()),
-    };
-    Ok(Sha256::digest(signature_payload.to_xdr(Limits::none())?).into())
+impl LocalKey {
+    pub fn new(key: ed25519_dalek::SigningKey, prompt: bool) -> Self {
+        Self { key, prompt }
+    }
+}
+
+#[async_trait::async_trait]
+impl Stellar for LocalKey {
+    async fn sign_blob(&self, data: &[u8]) -> Result<Vec<u8>, Error> {
+        if self.prompt {
+            eprintln!("Press 'y' or 'Y' for yes, any other key for no:");
```

Comment:
Does the prompt output any information about what is being signed, or just that something is being signed? What if multiple things need signing? Will it output this message a few times without context for each one?

Reply:
It prints each time something is signed, however, the context is currently just the public key and the hash. More context would be great, but as I was currently just following the way we were signing the hash, it was easiest to start with that context. Do you think it's worth adding to this PR? I saw it as a future improvement considering we don't even prompt for signing.